<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>665</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QGroupBox" name="groupBoxBind">
        <property name="title">
         <string>绑定</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>地址(&amp;A)：</string>
           </property>
           <property name="buddy">
            <cstring>bindAddr</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="bindAddr">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>0.0.0.0</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QCheckBox" name="cbAny">
           <property name="text">
            <string>Any</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>端口(&amp;P)：</string>
           </property>
           <property name="buddy">
            <cstring>bindPort</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="bindPort">
           <property name="suffix">
            <string/>
           </property>
           <property name="prefix">
            <string/>
           </property>
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
           <property name="value">
            <number>21778</number>
           </property>
           <property name="displayIntegerBase">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QCheckBox" name="cbRandom">
           <property name="text">
            <string>Random</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="3">
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>模式</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QCheckBox" name="cbShareAddress">
              <property name="text">
               <string>ShareAddress</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="cbDontShareAddress">
              <property name="text">
               <string>DontShareAddress</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="cbReuseAddressHint">
              <property name="text">
               <string>ReuseAddressHint</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>日志</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton">
             <property name="text">
              <string>清空</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPlainTextEdit" name="plainTextEditLog">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="maximumBlockCount">
            <number>100</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <item>
       <widget class="QGroupBox" name="groupBoxSend">
        <property name="title">
         <string>发送(&amp;S)</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="2" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>网卡(&amp;I)：</string>
           </property>
           <property name="buddy">
            <cstring>iFaceSend</cstring>
           </property>
          </widget>
         </item>
         <item row="4" column="1" colspan="2">
          <widget class="QSlider" name="sliderFrequency">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>10</number>
           </property>
           <property name="pageStep">
            <number>1</number>
           </property>
           <property name="value">
            <number>3</number>
           </property>
           <property name="sliderPosition">
            <number>3</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="invertedAppearance">
            <bool>false</bool>
           </property>
           <property name="invertedControls">
            <bool>false</bool>
           </property>
           <property name="tickPosition">
            <enum>QSlider::TicksBothSides</enum>
           </property>
           <property name="tickInterval">
            <number>1</number>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="dstAddr">
           <property name="text">
            <string>224.112.212.11</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>地址(&amp;A)：</string>
           </property>
           <property name="buddy">
            <cstring>dstAddr</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>端口(&amp;P)：</string>
           </property>
           <property name="buddy">
            <cstring>dstPort</cstring>
           </property>
          </widget>
         </item>
         <item row="3" column="0" colspan="2">
          <widget class="QCheckBox" name="cbMulticastLoopbackOption">
           <property name="text">
            <string>MulticastLoopbackOption</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>频率(&amp;F)：</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="dstPort">
           <property name="suffix">
            <string/>
           </property>
           <property name="prefix">
            <string/>
           </property>
           <property name="minimum">
            <number>10000</number>
           </property>
           <property name="maximum">
            <number>65535</number>
           </property>
           <property name="value">
            <number>21778</number>
           </property>
           <property name="displayIntegerBase">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="iFaceSend">
           <property name="enabled">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QCheckBox" name="cbInterface4Send">
           <property name="text">
            <string>指定网卡</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="5" column="0" colspan="3">
          <widget class="QPlainTextEdit" name="plainTextEditSend">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="maximumBlockCount">
            <number>100</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBoxRecv">
        <property name="title">
         <string>接收(&amp;R)</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QGroupBox" name="groupBoxJoin">
           <property name="title">
            <string>加组(&amp;M)</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="0" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>地址(&amp;A)：</string>
              </property>
              <property name="buddy">
               <cstring>multiAddr</cstring>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="multiAddr">
              <property name="text">
               <string>224.112.212.11</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>网卡(&amp;I)：</string>
              </property>
              <property name="buddy">
               <cstring>iFaceJoin</cstring>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="iFaceJoin"/>
            </item>
            <item row="1" column="2">
             <widget class="QCheckBox" name="cbInterface">
              <property name="text">
               <string>指定网卡</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>报文</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_2">
             <property name="text">
              <string>清空(&amp;C)</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPlainTextEdit" name="plainTextEditRecv">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="maximumBlockCount">
            <number>100</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="btnSwitch">
        <property name="text">
         <string>启动(&amp;B)</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>cbAny</sender>
   <signal>toggled(bool)</signal>
   <receiver>bindAddr</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>286</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>208</x>
     <y>86</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbRandom</sender>
   <signal>toggled(bool)</signal>
   <receiver>bindPort</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>286</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel">
     <x>208</x>
     <y>113</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbInterface</sender>
   <signal>toggled(bool)</signal>
   <receiver>iFaceJoin</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>690</x>
     <y>397</y>
    </hint>
    <hint type="destinationlabel">
     <x>613</x>
     <y>398</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbInterface4Send</sender>
   <signal>toggled(bool)</signal>
   <receiver>iFaceSend</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>305</x>
     <y>359</y>
    </hint>
    <hint type="destinationlabel">
     <x>228</x>
     <y>360</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>plainTextEditLog</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>614</x>
     <y>99</y>
    </hint>
    <hint type="destinationlabel">
     <x>577</x>
     <y>221</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>plainTextEditRecv</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>699</x>
     <y>438</y>
    </hint>
    <hint type="destinationlabel">
     <x>591</x>
     <y>551</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbRandom</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupBoxRecv</receiver>
   <slot>setHidden(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>286</x>
     <y>112</y>
    </hint>
    <hint type="destinationlabel">
     <x>393</x>
     <y>301</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnSwitch</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupBoxBind</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>721</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>288</x>
     <y>230</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnSwitch</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupBoxSend</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>705</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>440</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnSwitch</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupBoxRecv</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>751</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>535</x>
     <y>431</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbRandom</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupBoxRecv</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>271</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>377</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
